### PR TITLE
feat: Hook up Supabase Auth with server side cookie

### LINKED ISSUE #1: Set up Supabase Auth
# Chore

We need a couple of things for auth:

### Set up a login/register component

@MildTomato how soon do you think the component will be ready? If it's >2 days we can just custom code this or use [thor's](https://github.com/thorwebdev/nextjs-subscription-payments/blob/main/pages/signin.js) code

### Cookies

@thorwebdev is this one good to go? https://github.com/supabase/gotrue-js/pull/33

We'll need serverside validation for this one: https://github.com/supabase/next-server-components/blob/master/libs/session.js

https://github.com/supabase/next-server-components/blob/244e5e7c0f4aafdf0304d552e763fe6496b37bf7/pages/api/notes/index.js#L6

### PR DESCRIPTION
**Pull request by thorwebdev**
Closes #1 

## What kind of change does this PR introduce?

- Adds Supabase Auth with server side cookie helpers

## TODO

There are two issues:

1. The update doesn't seem to work, it does not update the note in the database.
2. The delete operation deletes all notes, not just the one it's supposed to

@kiwicopple would you be able to help investigate these two?

### CODE REVIEW

File: components/AuthButton.server.js [hunk 1]
```diff
@@ -1,21 +1,21 @@
-import React from 'react'
+import React, { useReducer } from 'react'
```
Comment (leerob) [inline components/AuthButton.server.js]: Is this needed?

**Review by leerob — COMMENTED** [review]
Nice! 💯

File: components/AuthButton.server.js (hunk 1 above)
Comment (thorwebdev) [reply components/AuthButton.server.js]: Looks like an accidental vscode auto import. Thanks for finding! :)

**Comment by kiwicopple** [comment]
this is awesome. thanks @thorwebdev

**Pull request merged**